src/display_backend.rs

```diff
@@ -1,11 +1,171 @@
+            // This absolutely change but in the meanwhile I'll keep it here to not forget
+            self.conn
+                .change_property(
+                    PropMode::Replace,
+                    win_id,
+                    self.atoms._NET_WM_STRUT,
+                    self.atoms.CARDINAL,
+                    32,
+                    4,
+                    &strut_list[0..16],
+                )?
+                .check()?;
+            self.conn
+                .change_property(
+                    PropMode::Replace,
+                    win_id,
+                    self.atoms._NET_WM_STRUT_PARTIAL,
+                    self.atoms.CARDINAL,
+                    32,
+                    12,
+                    &strut_list,
+                )?
+                .check()?;
+            self.conn.flush()?;
+            Ok(())
+        }
+    }
+        }
+
+        fn set_anchor(strut_def:StrutDefinition) {
+            let top=false;
+            let left=false;
+            let right=false;
+            let bottom=false;
+
+            match strut_def.side {
+                Edge::Top=>top=true,
+                Edge::Left=>left=true;
+                Edge::Right=>right=true,
+                Edge::Bottom=>bottom=true,
+                Edge::Center=>{},
+                Edge::Top_Left=>{
+                    top=true;
+                    left=true;
+                }
+                Edge::Top_Right=>{
+                    top=true;
+                    right=true;
+                }
+                Edge::Bottom_Right=>{
+                    bottom=true;
+                    right=true;
+                }
+                Edge::Bottom_Left=>{
+                    left=true;
+                    bottom=true;
+                }
+            }
+
+            // Anchors are if the window is pinned to each edge of the output
+            gtk_layer_shell::set_anchor(self.window, gtk_layer_shell::Edge::Left, top);
+            gtk_layer_shell::set_anchor(self.window, gtk_layer_shell::Edge::Right, right);
+            gtk_layer_shell::set_anchor(self.window, gtk_layer_shell::Edge::Top, top);
+            gtk_layer_shell::set_anchor(self.window, gtk_layer_shell::Edge::Bottom, bottom);
+        }
+
+        fn set_margin(margin:(u32,u32,u32,u32),window: &gtk::Window) {
+            let (margin_top, margin_right, margin_bottom, margin_left) = margin;
+
+            if margin_top>0 {
+                gtk_layer_shell::set_margin(window, gtk_layer_shell::Edge::Top, margin_top);
+            }
+            if margin_right>0 {
+                gtk_layer_shell::set_margin(window, gtk_layer_shell::Edge::Right, margin_right);
+            }
+            if margin_bottom>0 {
+                gtk_layer_shell::set_margin(window, gtk_layer_shell::Edge::Bottom, margin_bottom);
+            }
+            if margin_left>0 {
+                gtk_layer_shell::set_margin(window, gtk_layer_shell::Edge::Left, margin_left);
+            }
```

Comment:
why the if-checks? shouldn't this, in case you run it twice, once with a value > 0, then with 0, reset the margins of that side to 0?

Reply:
I didn't took into consideration theses values could be edited after run time. In this case the if aren't needed. If was just to not pointlessly set 0 margins.